bids/analysis/variables.py

```diff
@@ -155,13 +155,25 @@ def get_entities(img):
                 event_dfs.append(_df)
                 save_run = True
 
+        if extract_confounds:
+            sub_ents = {k: v for k, v in f_ents.items()
+                        if k not in ('type', 'unique_run_id')}
+            confound_files = layout.get(type='confounds', **sub_ents)
+            for cf in confound_files:
+                _data = pd.read_csv(cf.filename, sep='\t', na_values='n/a')
+                if columns is not None:
+                    conf_cols = list(set(_data.columns) & set(columns))
+                    _data = _data.loc[:, conf_cols]
+                if len(_data.columns) == 0:
+                    continue
+                confound_dfs.append(_data)
+                save_run = True
```

Comment:
So this right now doesn't resample, because the goal is to eventually end up with a design matrix where the confounds match the TRs, so upsampling/sparsifying and then downsampling/densifying doesn't make a lot of sense.

However, running this as is produces a mismatch in rows. Do you have thoughts on how to handle a situation like this?

Reply:
Is `type='confounds'` defined in one of the extensions? Not sure what format this is supposed to be in, and I don't recall seeing it anywhere. I think the only variable file format defined in the core BIDS spec I haven't implemented yet is `*_beh.json`. Where is 'confounds' coming from?

Reply:
Sorry, missed this. It's in [functional derivatives](https://docs.google.com/document/d/16CvBwVMAs0IMhdoKmlmcm3W8254dQmNARo-7HhE-lJU/edit#heading=h.mqkmyp254xh6). It's a TSV.

> The file must have the same number of rows as there are volumes in the corresponding raw _bold file.